File: problems/806-Number-of-Lines-To-Write-String/solution.go
func numberOfLines(widths []int, s string) []int {
    line, width := 1, 0
    for i := 0; i < len(s); {
        if width+widths[s[i]-'a'] > 100 {
            line++
            width = 0
        } else {
            width += widths[s[i]-'a']
            i++
        }
    }

    return []int{line, width}
}

File: problems/806-Number-of-Lines-To-Write-String/solution.java
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int line = 1, width = 0;
        for (int i = 0; i < s.length();) {
            if (width + widths[s.charAt(i) - 'a'] > 100) {
                line++;
                width = 0;
            } else {
                width += widths[s.charAt(i) - 'a'];
                i++;
            }
        }

        return new int[]{line, width};
    }
}

